.github/workflows/build_and_test.yml

```diff
@@ -97,15 +114,19 @@ jobs:
         run: cmake --build "$BUILD_DIR" -j
 
       - name: Test PcapPlusPlus
+        shell: bash
         run: |
-          ${{ matrix.python }} -m pip install -U pip
+          ${{ matrix.python }} -m venv ./venv
+          source ./venv/bin/activate
+          ${{ matrix.python }} -m pip install pip
           ${{ matrix.python }} -m pip install -r ci/run_tests/requirements.txt
           ${{ matrix.python }} ci/run_tests/run_tests.py --interface eth0 ${{ matrix.test-flags }}
 
       - name: Test Examples
+        shell: bash
```

Comment:
the default shell is `sh`, and to `source` the venv activator, we need bash.

Reply:
In `sh`, instead of using `source` you can use `.`:
```
. ./venv/bin/activate
```

https://unix.stackexchange.com/questions/748962/how-to-activate-python-venv-in-a-non-bash-shell

Reply:
I tried, but it doesn't work.

Reply:
okay, let me try again.

Reply:
ah, well, seems it works :)